Sources/ZIPFoundation/Entry+Zip64.swift

```diff
@@ -0,0 +1,114 @@
+                return 8
+            case .diskNumberStart:
+                return 4
+            }
+        }
+    }
+
+    var data: Data {
+        var headerID = self.headerID
+        var dataSize = self.dataSize
+        var uncompressedSize = self.uncompressedSize
+        var compressedSize = self.compressedSize
+        var relativeOffsetOfLFH = self.relativeOffsetOfLocalHeader
+        var diskNumberStart = self.diskNumberStart
+        var data = Data()
+        withUnsafePointer(to: &headerID, { data.append(UnsafeBufferPointer(start: $0, count: 1))})
+        withUnsafePointer(to: &dataSize, { data.append(UnsafeBufferPointer(start: $0, count: 1))})
+        if uncompressedSize != 0 || compressedSize != 0 {
+            withUnsafePointer(to: &uncompressedSize, { data.append(UnsafeBufferPointer(start: $0, count: 1))})
+            withUnsafePointer(to: &compressedSize, { data.append(UnsafeBufferPointer(start: $0, count: 1))})
+        }
+        if relativeOffsetOfLocalHeader != 0 {
+            withUnsafePointer(to: &relativeOffsetOfLFH, { data.append(UnsafeBufferPointer(start: $0, count: 1))})
+        }
+        if diskNumberStart != 0 {
+            withUnsafePointer(to: &diskNumberStart, { data.append(UnsafeBufferPointer(start: $0, count: 1))})
+        }
+        return data
+    }
+
+    init?(data: Data, fields: [Field]) {
+        let headerLength = 4
+        guard fields.reduce(0, { $0 + $1.size }) + headerLength == data.count else { return nil }
+        var readOffset = headerLength
+        func value<T>(of field: Field) throws -> T where T: BinaryInteger {
+            if fields.contains(field) {
+                defer {
+                    readOffset += MemoryLayout<T>.size
+                }
+                guard readOffset + field.size <= data.count else {
+                    throw Entry.EntryError.invalidDataError
+                }
+                return data.scanValue(start: readOffset)
+            } else {
+                return 0
+            }
+        }
+        do {
+            dataSize = data.scanValue(start: 2)
+            uncompressedSize = try value(of: .uncompressedSize)
+            compressedSize = try value(of: .compressedSize)
+            relativeOffsetOfLocalHeader = try value(of: .relativeOffsetOfLocalHeader)
+            diskNumberStart = try value(of: .diskNumberStart)
+        } catch {
+            return nil
+        }
+    }
+
+    static func scanForZip64Field(in data: Data, fields: [Field]) -> Entry.Zip64ExtendedInformation? {
+        guard !data.isEmpty else { return nil }
+        var offset = 0
+        var headerID: UInt16
+        var dataSize: UInt16
+        let extraFieldLength = data.count
+        while offset < extraFieldLength - 4 {
+            headerID = data.scanValue(start: offset)
+            dataSize = data.scanValue(start: offset + 2)
+            let nextOffset = offset + 4 + Int(dataSize)
```

Comment:
Would be great if we could use something more self-documenting for those magic numbers here:
Either `MemoryLayout.size(...)` if applicable or constants with descriptive names.